plaidml/bridge/openvino/ops/deformable_convolution.cpp

```diff
@@ -0,0 +1,290 @@
+  int coord_num = static_cast<int>(std::pow(2.0, rank - 2));
+  std::vector<edsl::Tensor> deformed_index_coord_vec(coord_num);
+  std::vector<edsl::Tensor> deformed_input_vec(coord_num);
+  for (auto i = 0; i < coord_num; ++i) {
+    std::vector<edsl::Tensor> concat_vec(rank - 2);
+    for (auto j = 0; j < rank - 2; ++j) {
+      int flag = (i / static_cast<int>(std::pow(2.0, rank - 2 - j - 1))) % 2;
+      concat_vec[j] = (flag == 0) ? deformed_index_vec_floor[j] : deformed_index_vec_ceil[j];
+    }
+    deformed_index_coord_vec[i] = op::concatenate(concat_vec, rank);
+    deformed_index_coord_vec[i] = edsl::cast(deformed_index_coord_vec[i], DType::INT32);
+    deformed_input_vec[i] = edsl::gather(I, deformed_index_coord_vec[i]).mode(edsl::GatherMode::ND).batchDims(2);
+  }
+  // Change the shape of deformed_index_ceil tensor and deformed_index_floor tensor to original shape.
+  deformed_index_reshape_dims.erase(deformed_index_reshape_dims.end() - 1);
+  for (auto i = 0; i < rank - 2; ++i) {
+    deformed_index_vec_ceil[i] = op::reshape(deformed_index_vec_ceil[i], make_tuple<int>(deformed_index_reshape_dims));
+    deformed_index_vec_floor[i] =
+        op::reshape(deformed_index_vec_floor[i], make_tuple<int>(deformed_index_reshape_dims));
+  }
+  // Get deformed input tensor with multi-linear interpolation.
+  for (auto i = 0; i < rank - 2; ++i) {
+    for (auto j = 0; j < static_cast<int>(std::pow(2.0, rank - 2 - i - 1)); ++j) {
+      deformed_input_vec[j] = deformed_input_vec[j] * (deformed_index_vec_ceil[i] - deformed_index_vec[i]) +
+                              deformed_input_vec[j + static_cast<int>(std::pow(2.0, rank - 2 - i - 1))] *
+                                  (1.0 - deformed_index_vec_ceil[i] + deformed_index_vec[i]);
+    }
+  }
+  edsl::Tensor deformed_input = deformed_input_vec[0];
+  // Set up the new strides.
+  std::vector<int> new_strides(F_shape.begin() + 2, F_shape.end());
+  // Compute DeformableConvolution.
+  edsl::Tensor result = op::convolution(deformed_input, F)
+                            .strides(new_strides)
+                            .autopad_mode(plaidml::op::AutoPadMode::VALID)
+                            .input_layout(plaidml::op::TensorLayout::NCX)
+                            .filter_layout(plaidml::op::TensorLayout::KCX)
+                            .autogroup_mode(plaidml::op::AutoGroupMode::EXPLICIT)
+                            .group_layout(plaidml::op::GroupLayout::IN_C)
+                            .group_layout(plaidml::op::GroupLayout::IN_K)
+                            .groups(G);
+  return result;
+}
+
+}  // namespace
+
+namespace PlaidMLPlugin {
+
+void registerDeformableConvolution() {
+  registerOp("DeformableConvolution", [](const Context& ctx) {
+    auto* layer = ngraph::as_type<ngraph::opset4::DeformableConvolution>(ctx.layer);
+    IE_ASSERT(ctx.operands.size() == 3);
+    auto I = ctx.operands.at(0);
+    auto OFF = ctx.operands.at(1);
```

Comment:
I wonder if `DS` meaning deformable shapes is better than using `OFF`

Reply:
https://github.com/Flex-plaidml-team/openvino/pull/37/files#diff-d3445805f5f95b68198f825b828da0b7660147ce36c1dffa5677c4d6a89c1ccdR25

Reply:
I think if change OFF to DS, there will be many variables to be changed, such as OFF_C, OFF_reshape_dims, OFF_transpose_dims and so on, maybe offset will be changed to deform_value(deformable values).

Reply:
Ok, I think OFF is ok. Maybe just add a comment that this is the deformable shapes in OV doc. I think offset is consistant the original paper though.

Reply:
what about "OFF means offset, and it is deformable values tensor in openvino doc" ?

Reply:
https://docs.openvinotoolkit.org/latest/openvino_docs_ops_convolution_DeformableConvolution_1.html

Reply:
I think it is good.